app_flutter/lib/service/google_authentication.dart

```diff
@@ -49,8 +49,17 @@ class GoogleSignInService {
   GoogleSignInAccount user;
 
   /// Authentication token to be sent to Cocoon Backend to verify API calls.
-  Future<String> get idToken => user?.authentication
-      ?.then((GoogleSignInAuthentication key) => key.idToken);
+  ///
+  /// If there is no currently signed in user, it will prompt the sign in
+  /// process before attempting to return an id token.
+  Future<String> get idToken async {
+    if (!await isAuthenticated) {
+      await signIn();
+    }
+
```

Comment:
Nit: assert idToken isn't null to make sure `signIn` function succeeded.

Reply:
Added